src/tests/ftest/control/dmg_pool_query_test.py

```diff
@@ -0,0 +1,112 @@
+    """Test Class Description:
+    Simple test to verify the pool query command of dmg tool.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DmgPoolQueryTest object."""
+        super(DmgPoolQueryTest, self).__init__(*args, **kwargs)
+        self.pool = None
+
+    def test_pool_query_basic(self):
+        """
+        JIRA ID: DAOS-2976
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Provided a valid pool UUID, verify the output received from
+        pool query command.
+        :avocado: tags=all,tiny,pr,dmg,pool_query,basic
+        """
+        self.pool = TestPool(self.context)
+        self.pool.get_params(self)
+        self.pool.create()
+        self.pool.connect(1 << 1)
+
+        # Update hostlist value for dmg command
+        port = self.params.get("port", "/run/server_config/*")
+        host_p = ["{}:{}".format(host, port) for host in self.hostlist_servers]
+
+        self.log.info("Running dmg pool query")
+        dmg_out = pool_query(self.bin, host_p, self.pool.uuid)
+
+        # Parse output
+        d_info = {}
+        d_info["dmg_t_cnt"] = re.findall(
+            r"Target\(VOS\) count:(.+)", dmg_out.stdout)
+        d_info["dmg_t_size"] = re.findall(r"Total size: (.+)", dmg_out.stdout)
+        d_info["dmg_mem_info"] = re.findall(
+            r"Free: (.+), min:(.+), max:(.+), mean:(.+)", dmg_out.stdout)
+        d_info["dmg_r_info"] = re.findall(
+            r"Rebuild (.+), (.+) objs, (.+) recs", dmg_out.stdout)
+
+        # TODO: Get data from API to verify dmg output.
+        e_info = {}
+        e_info["exp_t_cnt"] = None
+        e_info["exp_t_size"] = None
+        e_info["exp_mem_info"] = None
+        e_info["exp_r_info"] = None
+
+        # Verify
+        for k, e, d in zip(e_info.keys(), e_info.values(), d_info.values()):
+            if e != d:
+                self.fail("dmg pool query expected output: {}:{}".format(k, e))
+
+    def test_pool_query_inputs(self):
+        """
+        JIRA ID: DAOS-2976
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Verify the inputs that can be provided to 'query --pool'
+        argument of the dmg pool subcommand.
+        :avocado: tags=all,tiny,pr,dmg,pool_query,basic
+        """
+        # Get test UUID
+        expected_out = []
+        uuid = self.params.get("uuid", '/run/pool_uuids/*/')
+        expected_out.append(uuid[1])
+        self.log.info("Using test UUID: {}".format(uuid))
+
+        # Update hostlist value for dmg command
+        port = self.params.get("port", "/run/server_config/*")
+        host_p = ["{}:{}".format(host, port) for host in self.hostlist_servers]
+
+        self.log.info("Running dmg pool query")
+        dmg_out = pool_query(self.bin, host_p, uuid)
+
+        # Verify
+        self.log.info("Test expected to finish with: {} exit status".format(
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments